docs/docs/guides/operate/managing-concurrency.md

```diff
@@ -130,43 +522,360 @@ You can use Dagster's rich metadata to use a schedule or a sensor to only start
+```python
+import psutil
 
-<Tabs>
-  <TabItem value="Dagster+" label="Dagster+">
-    If runs aren't being dequeued in Dagster+, the root causes could be:
-    * **If using a [hybrid deployment](/deployment/dagster-plus/hybrid)**, the agent serving the deployment may be down. In this situation, runs will be paused.
-    * **Dagster+ is experiencing downtime**. Check the [status page](https://dagstercloud.statuspage.io/) for the latest on potential outages.
+def get_memory_based_concurrency():
+    available_gb = psutil.virtual_memory().available / (1024**3)
+    if available_gb > 16:
+        return 8
+    elif available_gb > 8:
+        return 4
+    else:
+        return 2
 
-  </TabItem>
-  <TabItem value="Dagster Open Source" label="Dagster Open Source">
-  If runs aren't being dequeued in Dagster Open Source, the root cause is likely an issue with the Dagster daemon or the run queue configuration.
+# Use in executor configuration
+```
+
+2. **Use resource-aware tags**:
+```python
+@asset(tags={"memory_usage": "high"})
+def memory_intensive_asset():
+    pass
+
+# Limit high memory usage assets
+job_config = {
+    "execution": {
+        "config": {
+            "multiprocess": {
+                "tag_concurrency_limits": [
+                    {"key": "memory_usage", "value": "high", "limit": 1}
+                ]
+            }
+        }
+    }
+}
+```
+
+#### Issue: Database connection pool exhaustion
+
+**Problem**: Too many concurrent database operations exhaust connection pool.
+
+**Solutions**:
+
+1. **Use database-specific concurrency limits**:
+```yaml
+concurrency:
+  runs:
+    tag_concurrency_limits:
+      - key: "database"
+        value: "postgres"
+        limit: 5  # Match your connection pool size
+      - key: "database"
+        value: "snowflake"
+        limit: 10
+```
+
+2. **Implement connection pool monitoring**:
+```python
+@asset(tags={"database": "postgres", "connection_type": "read_heavy"})
+def read_heavy_asset():
+    # Monitor connection usage
+    pass
+```
+
+### Monitoring and Debugging
+
+#### Debug concurrency configuration
+
+Use these approaches to debug concurrency issues:
+
+1. **Check daemon logs**:
+```bash
+# Look for concurrency-related messages
+grep -i "concurrency\|queue\|limit" /path/to/daemon.log
+```
+
+2. **Verify configuration in Dagster UI**:
+   - Navigate to **Deployment > Configuration**
+   - Check `concurrency` section for your settings
+   - Verify `run_queue` configuration
+
+3. **Monitor run queue status**:
+```python
+from dagster import DagsterInstance
+
+instance = DagsterInstance.get()
+queued_runs = instance.get_runs(filters=RunsFilter(statuses=[DagsterRunStatus.QUEUED]))
+print(f"Queued runs: {len(queued_runs)}")
+```
+
+4. **Check run tags and concurrency application**:
+```python
+# Verify tags are properly applied
+run = instance.get_run_by_id(run_id)
+print(f"Run tags: {run.tags}")
+```
 
-**Troubleshoot the Dagster daemon**
+### When runs remain in QUEUED status
```

Comment:
It looks like this might be a rewrite of the "Runs remaining in QUEUED status" section. If that's the case, I recommend leaving the existing troubleshooting section ("Runs remaining in QUEUED status") as is, since the rewrite seems to have removed important details.